files/image_config/ntp/ntp.conf.j2

```diff
@@ -41,13 +41,15 @@ interface ignore wildcard
 # if the source interface is configured but no ip on that interface, then listen on another
 # interface based on existing logic
 {%- macro check_ip_on_interface(interface_name, table_name) %}
+    {%- set ns = namespace(valid_intf = 'false') %}
     {%- if table_name %}
         {%- for (name, source_prefix) in table_name|pfx_filter %}
             {%- if source_prefix and name == interface_name %}
-true
+                {%- set ns.valid_intf = 'true' %}
             {%- endif %}
         {%- endfor %}
     {%- endif %}
+{{ ns.valid_intf }}
```

Comment:
Do you have to set ns.valid_intf to false as default value? What if the condition is not met?

Reply:
It is set to a default value of false here https://github.com/Azure/sonic-buildimage/pull/7586/files#diff-0a4c1f55aab272c1613693aea10e89dbab5b46a6dd65182f9139717693be421dR44. 

If the condition is not met (i.e. it never finds a valid ip on the interface) it simply returns false.